backend/server/game_api.js

```diff
@@ -0,0 +1,637 @@
+        }
+    };
+
+    if (count < gameShips[ship_size]) {
+        return true; //'can place'
+    } else {
+        return false; //'cannot place'
+    };
+};
+
+
+async function placeShip(game_id, user_id, ship_size, start_row, start_col, end_row, end_col, res) {
+    await pool.query('insert into ships (game_id, user_id, size, start_row, start_col, end_row, end_col) values ($1, $2, $3, $4, $5, $6, $7)', [game_id, user_id, ship_size, start_row, start_col, end_row, end_col]);
+    return res.status(200).json({msg: `Placed a ship of size ${ship_size}`});
+};
+
+
+gameRouter.post('/games/:game_id/place', async (req, res) => {
+    const { ship_size, start_row, start_col, end_row, end_col } = req.body;
+
+    if (start_row === end_row) {
+        if ((end_col - start_col + 1) !== ship_size) {
+            return res.status(400).json({ msg: 'Ship is not in the correct size' });
+        }
+    } else if (start_col === end_col) {
+        if ((end_row - start_row + 1) !== ship_size) {
+            return res.status(400).json({ msg: 'Ship is not in the correct size' });
+        }
+    } else {
+        return res.status(400).json({ msg: 'Ship is not valid' });
+    };
+
+    try {
+        const game = await pool.query('select * from games where id = $1', [req.params.game_id]);
+        const gameDetails = game.rows[0];
+
+        if (gameDetails.state === 'accepted' || (gameDetails.user1 === req.user.id && gameDetails.state === 'user2_ready') || (gameDetails.user2 === req.user.id && gameDetails.state === 'user1_ready')) {
+            if (req.user.id !== gameDetails.user1 && req.user.id !== gameDetails.user2) {
+                return res.status(400).json({ msg: 'User is not a player in the game' });
+            }
+
+            if (start_row > gameDetails.dimension || start_col > gameDetails.dimension || end_col > gameDetails.dimension || end_row > gameDetails.dimension) {
+                return res.status(400).json({ msg: 'Ship is not inside board borders' });
+            };
+
+            if (start_row < 1 || start_col < 1 || end_col < 1 || end_row < 1) {
+                return res.status(400).json({ msg: 'Ship is not inside board borders' });
+            };
+
+            const canPlace = await checkShips(req, gameDetails.dimension, ship_size);
+            if (!canPlace) {
+                return res.status(400).json({ msg: 'There are no more ships of this size to place' });
+            }
+
+
+            const ships = await pool.query('select * from ships where game_id = $1 and user_id = $2', [req.params.game_id, req.user.id]);
+            if (ships.rows.length === 0) {
+                await placeShip(req.params.game_id, req.user.id, ship_size, start_row, start_col, end_row, end_col, res);
+            } else {
+                const shipPlacementResult = ships.rows.map(ship => checkShipPlacement(start_row, end_row, start_col, end_col, ship)).reduce((accumulator, currentValue) => accumulator + currentValue, 0);
```

Comment:
instead of `map` and `reduce`, you can use `some` which checks if there is at least one element in the list that passes some test. 
See the explanation of this function here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some